src/callbacks/callbacks.jl

```diff
@@ -301,3 +301,25 @@ function gc_func(integrator)
     )
     return nothing
 end
+
+function maybe_graceful_exit(integrator)
+    output_dir = integrator.p.output_dir
+    file = joinpath(output_dir, "graceful_exit.dat")
+    if isfile(file)
+        open(file, "r") do io
+            while !eof(io)
+                try
+                    code = parse(Int, read(io, Char))
+                    return code != 0
+                catch
+                    open(io -> print(io, 0), file, "w")
+                    return false
+                end
```

Comment:
what does this do?

Reply:
If the file is somehow corrupted, this "resets" the file to a "continue running" state.

Reply:
why not just make it "if the file exists?". Then you can just do
```
touch outdir/exit
```

Reply:
(you would also want to remove the file if it exists at the start, i.e. if re-running the simulation)

Reply:
I think that reading a number from a file will open up more control options in the future. E.g., we can assign different actions to different numbers.

Reply:
Sure, I don't have strong opinions on this. I was only pointing out its very easy to create an empty file.

Reply:
Ok, I'll leave it as it is.